arch/x86/core/multiboot.c

```diff
@@ -11,4 +11,59 @@
 
 struct x86_multiboot_info x86_multiboot_info;
 
+#ifdef CONFIG_X86_MULTIBOOT_FRAMEBUF
+
+#include <display/framebuf.h>
+
+static struct framebuf_dev_data multiboot_framebuf_data = {
+	.width = CONFIG_X86_MULTIBOOT_FRAMEBUF_X,
+	.height = CONFIG_X86_MULTIBOOT_FRAMEBUF_Y
+};
+
+static int multiboot_framebuf_init(struct device *dev)
+{
+	struct framebuf_dev_data *data = FRAMEBUF_DATA(dev);
+	struct x86_multiboot_info *info = &x86_multiboot_info;
+
+	if ((info->flags & X86_MULTIBOOT_INFO_FLAGS_FB) &&
+	    (info->fb_width >= CONFIG_X86_MULTIBOOT_FRAMEBUF_X) &&
+	    (info->fb_height >= CONFIG_X86_MULTIBOOT_FRAMEBUF_Y) &&
+	    (info->fb_bpp == 32) && (info->fb_addr_hi == 0)) {
```

Comment:
Here the device config actually seems to be getting in the way.  We actually get the real dimensions at runtime, there's no reason to try to limit it to a particular compile-time size, the memory for the framebuffer has already been "allocated" and it's ready to use, right?  So why not just populate the width/height values with the ones from the firmware and skip the kconfig layer entirely?

Reply:
I think the the application should be able to determine the dimensions of the framebuffer it wants at compile time, which is why they are Kconfig values -- this isn't Windows, I don't think embedded applications should be forced to adapt dynamically to the display size.  

The problem is that, while these dimensions are passed down to the multiboot loader in crt0.S, the loader is not required to honor the request (or it might not be able to).  So my logic was, as long as the framebuffer is at least big enough, pretend it's the right size and center it on-screen.